internal/controllers/addon_controller.go

```diff
@@ -53,14 +56,19 @@ func (r *AddonReconciler) Reconcile(
 			return ctrl.Result{}, nil
 		}
 
-		return ctrl.Result{}, r.reportTerminationStatus(ctx, addon)
+		err := r.reportTerminationStatus(ctx, addon)
+		if err != nil {
+			log.Error(err, "failed to update termination status")
```

Comment:
we should probably:
```suggestion
			return ctrl.Result{}, fmt.Errorf("failed to update termination status: %w", err)
```

Same with other error logs, controller-runtime will already log any errors returned by `.Reconcile` and I would try to avoid to log the same error multiple times. With `%w` we also allow to unwrap errors and check the root cause, so we can handle errors further up in the call stack. And handled errors should not appear in the log at all IMHO.

Reply:
fixed